packages/extension-base/src/background/handlers/Extension.ts

```diff
@@ -137,20 +133,39 @@ export default class Extension {
     return true;
   }
 
-  private refreshAccountPasswordCache (pair: KeyringPair): number {
+  private async refreshAccountPasswordCache (pair: KeyringPair): Promise<{remainingTime: number, cachedPassword?: string}> {
     const { address } = pair;
 
-    const savedExpiry = this.#cachedUnlocks[address] || 0;
-    const remainingTime = savedExpiry - Date.now();
+    const savedPasswords = await chromeStorage.getItem('password-cache') ?? {};
+    const expiresAt = savedPasswords[address]?.expiresAt ?? 0;
+    const remainingTime = expiresAt - Date.now();
+    const cachedPassword = savedPasswords[address]?.password;
 
-    if (remainingTime < 0) {
-      this.#cachedUnlocks[address] = 0;
-      pair.lock();
+    if (remainingTime > 0) {
+      return { remainingTime, cachedPassword };
+    }
 
-      return 0;
+    if (address in savedPasswords) {
+      delete savedPasswords[address];
+      await chromeStorage.setItem('password-cache', () => (savedPasswords));
     }
 
-    return remainingTime;
+    pair.lock();
+
+    return { remainingTime: 0 };
+  }
+
+  private async clearPasswordCache (pair: KeyringPair): Promise<void> {
+    const { address } = pair;
+
+    const savedPasswords = await chromeStorage.getItem('password-cache') ?? {};
+
+    if (address in savedPasswords) {
+      delete savedPasswords[address];
+      await chromeStorage.setItem('password-cache', () => (savedPasswords));
+    }
```

Comment:
we could use this method to simplify `refreshAccountPasswordCache`
```suggestion
    await chromeStorage.setItem('password-cache', ({ [address]: _, ...savedPasswords } = {}) => (savedPasswords));
```

Reply:
eslint doesn't like the unused var :cry: , setting some ignore var pattern in eslint config would solve it

Reply:
https://eslint.org/docs/latest/rules/no-unused-vars#varsignorepattern

Reply:
I think first method may be a bit safer, since it clears out any mentions of address as well? or wdyt?

Reply:
> eslint doesn't like the unused var

`_` usually works.

Reply:
Not sure what you mean - `savedPasswords` contains all keys from cache, but the `address` which is assigned to `_`. Anyway, it's not needed, resolving